File: fixtures/scale/dart/medium/lib/memory/memory_bank.dart
import 'memory_slot.dart';

class MemoryBank {
  final Map<String, MemorySlot> _slots = {};

  MemorySlot slot(String name) => _slots.putIfAbsent(name, () => MemorySlot());

  void add(String name, double amount) {
    final target = slot(name);
    target.store(target.recall() + amount);
  }

  void clearAll() {
    for (final s in _slots.values) {
      s.clear();
    }
  }

  int occupiedCount() => _slots.values.where((s) => s.isOccupied).length;
}

File: fixtures/scale/scala/medium/src/calc/memory/MemoryBank.scala
package calc.memory

import scala.collection.mutable

final class MemoryBank {
  private val slots = mutable.Map.empty[String, MemorySlot]

  def slot(name: String): MemorySlot =
    slots.getOrElseUpdate(name, new MemorySlot)

  def add(name: String, amount: Double): Unit = {
    val target = slot(name)
    target.store(target.recall + amount)
  }

  def clearAll(): Unit = slots.values.foreach(_.clear())

  def occupiedCount: Int = slots.values.count(_.isOccupied)
}
